### ISSUE TITLE
Escape mechanism doesn't work in like/ilike functions

### ISSUE THREAD

**Issue by bziobrowski** [questdb/questdb#2623]
### Describe the bug

Like and ilike functions don't recognize default escape character '\' nor allow using a different one via ESCAPE clause . 

### To reproduce

```sql
SELECT '\quest' LIKE '\_uest'; 
--true but should be false

SELECT 'quest_' LIKE 'quest\_'
--false but should be true 

select 'quest' like 'quest' escape 'Z'
--found [tok=''Z'', len=3] ',', 'from' or 'over' expected
```

### Environment

```markdown
- **QuestDB version**:6.5.4/master
- **OS**:
- **Browser**:
```

**Comment by sharmadhiraj86** [comment]
Hi. I would like to contribute to this issue. Can you assign this issue to me? @bziobrowski

**Comment by Ln11211** [comment]
> ### Describe the bug
> Like and ilike functions don't recognize default escape character '' nor allow using a different one via ESCAPE clause .
> 
> ### To reproduce
> ```sql
> SELECT '\quest' LIKE '\_uest'; 
> --true but should be false
> 
> SELECT 'quest_' LIKE 'quest\_'
> --false but should be true 
> 
> select 'quest' like 'quest' escape 'Z'
> --found [tok=''Z'', len=3] ',', 'from' or 'over' expected
> ```
> 
> ### Expected Behavior
> _No response_
> 
> ### Environment
> ```gfm
> - **QuestDB version**:6.5.4/master
> - **OS**:
> - **Browser**:
> ```
> 
> ### Additional context
> _No response_

Yeah @bziobrowski  , I can confirm it. I just tested the escape sequences and they don't seem to work, it has something to do with how we handle the char sequence. 

![image](https://user-images.githubusercontent.com/91385411/195573356-3d8cff17-9826-447d-a199-88621acf605e.png)

**Comment by onlydevelop** [comment]
@sharmadhiraj86 did you make any progress in this?

**Comment by shivambarot** [comment]
@bziobrowski can you please assign me this task?

**Comment by KACHIDA** [comment]
Hi @bziobrowski , I'm interested to work on the above issue, can you please assign me this task

**Comment by bziobrowski** [comment]
I've cleared the assignee field because I  haven't heard from the person . 
If there's no linked PR then it usually means that no one is working on the issue .

**Comment by onlydevelop** [comment]
@bziobrowski Can you give some starting point for this?

**Comment by bziobrowski** [comment]
LIKE is implemented with 
io.questdb.griffin.engine.functions.regex.ILikeStrFunctionFactory and
io.questdb.griffin.engine.functions.regex.LikeStrFunctionFactory classes . 
To support ESCAPE clause you'd need to change io.questdb.griffin.ExpressionParser (see how cast or between operators are implemented ) and then pass the optional escape character to functions mentioned above .

**Comment by SiyaoIsHiding** [comment]
Our team is working on QuestDB in our Testing & Debugging coursework and we wanna fix this issue. May u assign this issue to me?

**Comment by bziobrowski** [comment]
@SiyaoIsHiding AFAIK no on is working on the issue so feel free to submit PR .

**Comment by AdalyatNazirov** [comment]
@bziobrowski is this issue still open? I can see the PR have been merged years ago, but issue still open

**Comment by maneeshchamakura** [comment]
Hi @SiyaoIsHiding this looks interesting. If the issue is open, can I work on this?

**Comment by manascb1344** [comment]
Hi can i work on this @bziobrowski ??
